splunk/entry_crud.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2022 Splunk, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package splunk
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/splunk/go-splunk-client/pkg/client"
+	"github.com/splunk/terraform-provider-splunk/internal/sync"
+)
+
+// useLegacyClient returns:
+//
+// * the resource's use_legacy_client value, if false or explicitly true
+//
+// * the provider's use_legacy_client_default value
+func useLegacyClient(provider *SplunkProvider, d *schema.ResourceData) bool {
+	resourceLegacyClientI, ok := d.GetOk("use_legacy_client")
+	resourceLegacyClient := resourceLegacyClientI.(bool)
+	// GetOk only returns true if the fetched value is not the zero value for its type,
+	// so we can only determine if use_legacy_client was explicitly true. but because
+	// true is our default value, we know that it can only be false if explicitly set.
+	if ok || !resourceLegacyClient {
+		return resourceLegacyClient
+	}
```

Comment:
I'm working to validate this comment and logic is correct, after behavior I saw while testing similar code yesterday.